src/cholesky.jl

```diff
@@ -1,21 +1,50 @@
 # Generic Cholesky decomposition for fixed-size matrices, mostly unrolled
 non_hermitian_error() = throw(LinearAlgebra.PosDefException(-1))
-@inline function LinearAlgebra.cholesky(A::StaticMatrix)
-    ishermitian(A) || non_hermitian_error()
+@inline function LinearAlgebra.cholesky(A::StaticMatrix; check=true)
+    !ishermitian(A) && (check && non_hermitian_error())
```

Comment:
```suggestion
    check && (!ishermitian(A) && non_hermitian_error())
```
This way if `check` is false then the test whether `A` is Hermitian isn't performed.

Reply:
Improved this so that it first checks if the matrix is Hermitian and then skips computing the factorization if it fails. Also eliminated duplicate code by calling the Hermitian method.

Reply:
Thanks, right, it looks better now.